### PR TITLE
Improving 'key search --long-list' to use tabwriter

### LINKED ISSUE #4156: Improve 'key search --long-list' to use text/tabwriter
## Version of Singularity:

```
3.3.0
```

### Expected behavior

```
$ singularity key search --long-list
```
Will use the `text/tabwriter` package.

### Actual behavior

Does not exist yet.

### Steps to reproduce behavior

Write here.

### PR DESCRIPTION
**Pull request by WestleyK**
## Description of the Pull Request (PR):

This PR improves the `key search --long-list` by using the `tabwriter` package. And refactor the mr parsing.

### This fixes or addresses the following GitHub issues:

- Fixes #4156

<br>

### CODE REVIEW

File: pkg/sypgp/sypgp.go [hunk 1]
```diff
@@ -703,112 +702,108 @@ func getEncryptionAlgorithmName(n string) (string, error) {
...
+				keyDateExpired, err = date(nk[5])
+				if err != nil {
+					panic(err)
+				}
+				if nk[6] == "r" {
+					keyStatus = "[revoked]"
+				} else if nk[6] == "d" {
+					keyStatus = "[disabled]"
+				} else if nk[6] == "e" {
+					keyStatus = "[xpired]"
```
Comment (gvallee) [inline pkg/sypgp/sypgp.go]:
```suggestion
					keyStatus = "[expired]"
```

**Review by gvallee — CHANGES_REQUESTED** [review]
I only have a very minor comment.

File: pkg/sypgp/sypgp.go [hunk 2]
```diff
@@ -703,112 +702,108 @@ func getEncryptionAlgorithmName(n string) (string, error) {
...
-		if err != nil {
-			return "", err
-		}
+	key := strings.Split(mrString, "\n")
 
-		expirationTimestamp, err := date(expiryDate)
-		if err != nil {
-			return "", err
-		}
+	var (
```
Comment (ikaneshiro) [inline pkg/sypgp/sypgp.go]: Why declare all of these variables here when they could be within the scope of the first loop? Then you would not need to "zero out" their values after iteration on the second loop completes.

File: pkg/sypgp/sypgp.go [hunk 3]
```diff
@@ -703,112 +702,108 @@ func getEncryptionAlgorithmName(n string) (string, error) {
...
+		keyReady       bool
+	)
 
-		// check the status of each key if flags are present
-		if len(featuresKey) == 7 {
-			if featuresKey[6] == "r" {
-				status = "revoked"
+	for _, k := range key {
+		nk := strings.Split(k, ":")
+		for _, n := range nk {
```
Comment (ikaneshiro) [inline pkg/sypgp/sypgp.go:796]: Could we break this inner loop into a function that returns the key parameters in a struct or something like that?

**Review by ikaneshiro — CHANGES_REQUESTED** [review]

File: pkg/sypgp/sypgp.go (hunk 2 above)
Comment (WestleyK) [reply pkg/sypgp/sypgp.go]: That wont work, the second loop can loop 1-3 times, and we dont know how much until after the `uid` section.

**Review by gvallee — APPROVED** [review]
lgtm

**Review by ikaneshiro — APPROVED** [review]

**Pull request merged**